### PR TITLE
Provide options for restricting directory search to people and/or groups (2.2)

### PR DESCRIPTION
**Pull request by kunklejr**
- [x] @joeferner
- [x] @kunklejr @mwizeman @sfeng88 @diegogrz
- [x] @dsingley @EvanOxfeld 
- [x] @joeybrk372 @rygim @jharwig 

CHANGELOG
Changed: Provide options for restricting directory search to people and/or groups

### CODE REVIEW

File: web/war/src/main/webapp/js/data/web-worker/services/directory.js [hunk 1]
```diff
@@ -14,10 +14,18 @@ define([
...
+            options = options || {};
+            var params = {
                 search: search
-            });
+            };
+            if (typeof options.people !== undefined) {
+                params.people = options.people;
+            }
+            if (typeof options.groups !== undefined) {
+                params.groups = options.groups;
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/data/web-worker/services/directory.js]: `typeof` returns a string. While you could remove the `typeof` to make it work, instead I'd recommend replacing the ifs with `_.extend` to copy any properties from the `options` object over to `params`:

```
search: function(search, options) {
  var params = _.extend({}, options, { search: search });
  return ajax('GET', '/directory/search', params);
}
```

Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/data/web-worker/services/directory.js]: @jharwig Is ES6 syntax supported here? We could simplify the above even further.

Comment (kunklejr) [inline web/war/src/main/webapp/js/data/web-worker/services/directory.js]: Two lines seems pretty simple to me. That's a nice change.

**Comment by kunklejr** [comment]
@EvanOxfeld: Your requested change was pushed.

File: web/war/src/main/webapp/js/data/web-worker/services/directory.js [hunk 2]
```diff
@@ -14,10 +14,9 @@ define([
             });
         },
 
-        search: function(search) {
-            return ajax('GET', '/directory/search', {
-                search: search
-            });
+        search: function(search, options) {
+            var params = _.extend({}, options, { search: search });
```
Comment (jharwig) [inline web/war/src/main/webapp/js/data/web-worker/services/directory.js]: can now do `var params = { ...options, search };`

File: web/war/src/main/webapp/js/data/web-worker/services/directory.js (hunk 1 above)
Comment (jharwig) [inline web/war/src/main/webapp/js/data/web-worker/services/directory.js]: Yes, see comment above

**Pull request merged**